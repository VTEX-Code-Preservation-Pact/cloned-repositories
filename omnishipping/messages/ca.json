{
  "omnishipping.deliveryTitle": "Entrega",
  "omnishipping.loading": "Carrgant…",
  "omnishipping.awatingData": "Encara falten dades per completar",
  "omnishipping.deliveryAddressTitle": "Completa la teva adreça d'entrega",
  "omnishipping.paymentCTA": "Anar al pagament",
  "omnishipping.locatePickupPoint": "Cercar un punt",
  "omnishipping.deliver": "Rebre",
  "omnishipping.pickup": "Recollir",
  "omnishipping.deliverySummary": "Rebre",
  "omnishipping.package": "Paquet",
  "omnishipping.items": "{itemsAmount, plural, =0 {} one {# producte} other {# productes}}",
  "omnishipping.pickupSummary": "Recollida",
  "omnishipping.cancelBackList": "Consultar altres punts de recollida",
  "omnishipping.cancelPickup": "Cancel·lar la recollida",
  "omnishipping.deliveryOption": "Opcions d'entrega",
  "omnishipping.pickupOption": "Forma de retirada",
  "omnishipping.chooseDelivery": "Rebre",
  "omnishipping.choosePickup": "Recollida a",
  "omnishipping.findPickup": "Trobar un punt de recollida proper.",
  "omnishipping.noPickupSelected": "No hi ha una botiga seleccionada",
  "omnishipping.pickupReceiver": "Responsable de la retirada",
  "omnishipping.slaOption": "{name} - {price} - {shippingEstimate}",
  "omnishipping.slaOptionSimple": "{shippingEstimate}{scheduled}",
  "omnishipping.shippingEstimateScheduled": "Data programada",
  "omnishipping.shippingEstimate-bd": "{timeAmount, plural, =0 {El mateix dia} one {En # dia hábil} other {Fins a # dies hàbils}}",
  "omnishipping.shippingEstimate-db": "Dies laborables",
  "omnishipping.shippingEstimate-d": "{timeAmount, plural, =0 {El mateix dia} one {En # dia} other {Fins a # dies}}",
  "omnishipping.shippingEstimate-h": "{timeAmount, plural, =0 {Amb recollida immediata} one {En # hora} other {Fins a # hores}}",
  "omnishipping.shippingEstimate-m": "{timeAmount, plural, =0 {Amb recollida immediata} one {En # minut} other {Fins a # minuts}}",
  "omnishipping.free": "Gratis",
  "omnishipping.shippingEstimatePickup-bd": "{timeAmount, plural, =0 {A punt en el mateix dia} one {A punt fins # dies hàbils} other {Listo en hasta # días hábiles}}",
  "omnishipping.shippingEstimatePickup-d": "{timeAmount, plural, =0 {A punt en el mateix dia} one {A punt fins # dia} other {A punt a fins # dies}}",
  "omnishipping.shippingEstimatePickup-h": "{timeAmount, plural, =0 {Amb recollida immediata} one {A punt fins # hora} other {A punt a fins # hores}}",
  "omnishipping.shippingEstimatePickup-m": "{timeAmount, plural, =0 {Amb recollida immediata} one {A punt fins # minut} other {A punt a fins # minuts}}",
  "omnishipping.scheduled": "Programada",
  "omnishipping.scheduledDeliveryChoice": "Escull una data d'entrega",
  "omnishipping.scheduledDeliverySplitDisclaimer": "Alguns Itenes no es poden programar. Seleccioneu una forma de lliurament per a rebre'ls.",
  "omnishipping.scheduledDeliveryPartial": "{amount, plural, =0 { } one {Programar # product} other {Programar # productes}}",
  "omnishipping.scheduledDeliveryPrice": "A partir de {price}",
  "omnishipping.scheduledDeliveryAll": "Programar tots",
  "omnishipping.deliveryOptionsLeft": "Forma d'entrega per als productes restants",
  "omnishipping.selectPickupPoint": "Esculli un punt de recollida",
  "omnishipping.selectDeliveryOptions": "Seleccioneu les formes d'entrega",
  "omnishipping.pointDetails": "Detalls del punt de recollida",
  "omnishipping.closeButton": "Tancar",
  "omnishipping.deliveryDate": "Programació",
  "omnishipping.deliveryTime": "Hora d'entrega",
  "omnishipping.datepickerDateFormat": "dddd, LL",
  "omnishipping.deliveryWindows": "Des de {startDate} fins {endDate}",
  "omnishipping.deliveryWindowsSummary": "{date}, des de les {startDate} fins {endDate}",
  "omnishipping.deliveryWindowsSummaryDate": "{date}, Entre {startDate} i {endDate}",
  "omnishipping.deliveryWindowsSummarytime": "Des de {startDate} fins a {endDate}",
  "omnishipping.list": "Llista",
  "omnishipping.map": "Mapa",
  "omnishipping.pickupPoint.modify": "Canviar",
  "omnishipping.modify": "canviar",
  "omnishipping.modifyPostalCode": "canviar",
  "omnishipping.details": "Veure més detalls",
  "omnishipping.change": "Veure tots els punts de recollida",
  "omnishipping.searchLocationMap": "Cercar per Lloc, Codi Postal, Barri o Ciutat",
  "omnishipping.productsInPoint": "Productes disponibles en aquest punt",
  "omnishipping.aditionalInfo": "Altres informacions",
  "omnishipping.unavailable": "No disponible",
  "omnishipping.unavailableItemsAmount": "{itemsAmount, plural, =0 {} one {# No disponible} other {# no estan disponibles}}",
  "omnishipping.confirmPoint": "Recollida en aquest punt",
  "omnishipping.modifyDeliveryOptions": "Canviar opcions d'entrega",
  "omnishipping.shippingAddress": "Adreça d'entrega",
  "omnishipping.editSelectedAddressButton": "Modificar l'adreça seleccionada",
  "omnishipping.addDeliveryPlaceButton": "Entregar en una altra adreça",
  "omnishipping.unavailableDelivery": "{items, plural, =0 {} one {El següent producte està no té entrega disponible per al codi postal escollit.} other {Els següents productes no tenen entrega disponible per al codi postal escollit.}}",
  "omnishipping.unavailableDeliveryAvailablePickup": "{items, plural, =0 {} one {El següent element no està disponible per al lliurament. Podeu treure-ho en un moment.} other {Els següents elements no estan disponibles per al lliurament. Podeu treure-les en un punt.}}",
  "omnishipping.unavailableDeliveryOptionsPostalCode": "Canvieu el codi postal o {removeFromCart}",
  "omnishipping.unavailableDeliveryOptions": "Seleccioneu un altre punt de recollida o {removeFromCart}",
  "omnishipping.changePostalCode": "Canvieu el seu codi postal",
  "omnishipping.optionPackagesDetails": "{items, plural, =0 {...} one {{shippingEstimate}} other {{items} paquets en diferents terminis}}",
  "omnishipping.onePackageIn": "Un paquet a",
  "omnishipping.removeFromCart": "{items, plural, =0 {} one {retira el producte de la cistella.} other {retira els productes de la cistella.}}",
  "omnishipping.pickUpItems": "{items, plural, =0 {} one {Retira l'element en un punt} other {Retira els elements en un punt}}",
  "omnishipping.allItemsPickup": "Només hi ha disponible l'opció Take Out per als vostres articles.",
  "omnishipping.noTimesAvailable": "No hi ha horari disponible per a aquesta data.",
  "omnishipping.productsFromSeller": "Productes de {seller}",
  "omnishipping.fillDeliveryData": "Ompliu el lliurament",
  "omnishipping.deliverySplitDisclaimer": "Alguns productes no poden ser entregats en aquesta direcció. Seleccioneu un punt de recollida.",
  "omnishipping.deliverySplitDisclaimerLogged": "Alguns productes no estan disponibles per a la seva direcció. Recull en un punt.",
  "omnishipping.pickupSplitDisclaimer": "Alguns productes no estan disponibles en el punt seleccionat. Introduïu la vostra adreça per rebre'ls.",
  "omnishipping.pickupSplitDisclaimerLogged": "Alguns productes no estan disponibles en el punt seleccionat. Introduïu la vostra adreça per rebre'ls.",
  "omnishipping.price": "{value, plural, =0 {Gratis} other {{formattedPrice}}}",
  "omnishipping.of": "de",
  "omnishipping.askGeolocation": "Trobi una adreça amb la seva ubicació",
  "omnishipping.cheapestOption": "Més econòmica",
  "omnishipping.fastestOption": "Més ràpida",
  "omnishipping.combinedOption": "Combinada",
  "omnishipping.yourPackages": "Seus paquets",
  "omnishipping.pickupSomeItems": "1. Seleccioneu Opcions de retirada",
  "omnishipping.receiveRemainingPackages": "2. Rebre els productes restants",
  "omnishipping.askGeolocationTitle": "Suggerirem un punt de recollida a prop seu",
  "omnishipping.askGeolocationSubtitle": "Per això, necessitem saber la seva ubicació",
  "omnishipping.askGeolocationAccept": "Utilitza la meva ubicació actual",
  "omnishipping.askGeolocationManual": "Cercar un punt manualment",
  "omnishipping.askGeolocationDenied": "Cercar per una adreça",
  "omnishipping.askGeolocationDeniedTitle": "Trobi punts de recollida propers",
  "omnishipping.askGeolocationDeniedSubtitle": "Busqui les adreces que vostè freqüenta i localitzarem els punts propers",
  "omnishipping.askGeolocationWaiting": "Esperant la seva autorització",
  "omnishipping.askGeolocationInstructions": "Seleccioneu \" Permet \" en l'alerta que apareix al seu navegador",
  "omnishipping.askGeolocationSearching": "Buscant la seva ubicació",
  "omnishipping.backToAddressList": "Tornar a la llista d'adreces",
  "omnishipping.jsError": "Ha succeït un conflicte inesperada",
  "omnishipping.reloadThePage": "Carregar la pàgina",
  "omnishipping.errorId": "ErrorId: {errorId}",
  "omnishipping.noAddressTitle": "Dades de l'adreça incompletes",
  "omnishipping.noAddressCTA": "Per introduir una adreça, {action}.",
  "omnishipping.noAddressAction": "Confirmi la seva identitat",
  "omnishipping.deliverComplement": "a casa",
  "omnishipping.pickupComplement": "a una botiga"
}
