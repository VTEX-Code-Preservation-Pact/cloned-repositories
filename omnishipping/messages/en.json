{
  "omnishipping.deliveryTitle": "Shipping",
  "omnishipping.loading": "Loadingâ€¦",
  "omnishipping.awatingData": "Waiting for more information",
  "omnishipping.deliveryAddressTitle": "Complete your address information",
  "omnishipping.paymentCTA": "Go to payment",
  "omnishipping.locatePickupPoint": "Locate a pickup point",
  "omnishipping.deliver": "Ship",
  "omnishipping.deliverComplement": "to address",
  "omnishipping.pickup": "Pick up",
  "omnishipping.pickupComplement": "in point",
  "omnishipping.deliverySummary": "Shipping to address",
  "omnishipping.package": "Package",
  "omnishipping.items": "{itemsAmount, plural, =0 {} one {# item} other {# items}}",
  "omnishipping.pickupSummary": "Pick up in a point",
  "omnishipping.cancelBackList": "See other pickup points",
  "omnishipping.cancelPickup": "Cancel pickup",
  "omnishipping.chooseDelivery": "Delivery",
  "omnishipping.choosePickup": "Package and delivery date",
  "omnishipping.deliveryOption": "Shipping method",
  "omnishipping.pickupOption": "Pickup method",
  "omnishipping.findPickup": "Find a pickup point next to you.",
  "omnishipping.noPickupSelected": "You haven't selected a store yet",
  "omnishipping.pickupReceiver": "Pickup receiver",
  "omnishipping.slaOption": "{name} - {price} - {shippingEstimate}",
  "omnishipping.slaOptionSimple": "{shippingEstimate}{scheduled}",
  "omnishipping.shippingEstimateScheduled": "Scheduled date",
  "omnishipping.shippingEstimate-bd": "{timeAmount, plural, =0 {Same day} one {In # business day} other {In up to # business days}}",
  "omnishipping.shippingEstimate-d": "{timeAmount, plural, =0 {Same day} one {In # day} other {In up to # days}}",
  "omnishipping.shippingEstimate-h": "{timeAmount, plural, =0 {Now} one {In # hours} other {In up to # hours}}",
  "omnishipping.shippingEstimate-m": "{timeAmount, plural, =0 {Now} one {In # minute} other {In up to # minutes}}",
  "omnishipping.free": "Free",
  "omnishipping.shippingEstimatePickup-bd": "{timeAmount, plural, =0 {Ready in the same day} one {Ready in up to # business day} other {Ready in up to # business days}}",
  "omnishipping.shippingEstimatePickup-d": "{timeAmount, plural, =0 {Ready in the same day} one {Ready in up to # day} other {Ready in up to # days}}",
  "omnishipping.shippingEstimatePickup-h": "{timeAmount, plural, =0 {Ready for pickup} one {Ready in up to # hour} other {Ready in up to # hours}}",
  "omnishipping.shippingEstimatePickup-m": "{timeAmount, plural, =0 {Ready for pickup} one {Ready in up to # minute} other {Ready in up to # minutes}}",
  "omnishipping.scheduled": "Scheduled",
  "omnishipping.scheduledDeliveryChoice": "Choose your shipping date",
  "omnishipping.scheduledDeliverySplitDisclaimer": "Some items can not be scheduled. Select a delivery method to receive them.",
  "omnishipping.scheduledDeliveryPartial": "{amount, plural, =0 { } one {Schedule # item} other {Schedule # items}}",
  "omnishipping.scheduledDeliveryAll": "Schedule all",
  "omnishipping.scheduledDeliveryPrice": "Rates from {price}",
  "omnishipping.deliveryOptionsLeft": "Delivery method for remaining items",
  "omnishipping.selectPickupPoint": "Select a pickup point",
  "omnishipping.selectDeliveryOptions": "Select delivery options",
  "omnishipping.pointDetails": "Pickup point details",
  "omnishipping.closeButton": "close",
  "omnishipping.deliveryDate": "Scheduling",
  "omnishipping.deliveryTime": "Delivery date",
  "omnishipping.datepickerDateFormat": "dddd, LL",
  "omnishipping.deliveryWindows": "From {startDate} to {endDate}",
  "omnishipping.deliveryWindowsSummary": "{date}, between {startDate} and {endDate}",
  "omnishipping.deliveryWindowsSummarytime": "between {startDate} and {endDate}",
  "omnishipping.list": "List",
  "omnishipping.map": "Map",
  "omnishipping.pickupPoint.modify": "Modify",
  "omnishipping.modify": "modify",
  "omnishipping.modifyPostalCode": "modify",
  "omnishipping.details": "See more details",
  "omnishipping.change": "See all pickup points available",
  "omnishipping.searchLocationMap": "Search for Street, ZIP Code, neighborhood or city",
  "omnishipping.productsInPoint": "Products available in this pickup point",
  "omnishipping.aditionalInfo": "Other information",
  "omnishipping.unavailable": "unavailable",
  "omnishipping.unavailableItemsAmount": "{itemsAmount, plural, =0 {} one {# item unavailable} other {# item unavailable}}",
  "omnishipping.confirmPoint": "Pickup in this point",
  "omnishipping.modifyDeliveryOptions": "Change shipping options",
  "omnishipping.shippingAddress": "Shipping address",
  "omnishipping.editSelectedAddressButton": "Update selected address information",
  "omnishipping.addDeliveryPlaceButton": "Deliver in another address",
  "omnishipping.unavailableDelivery": "{items, plural, =0 {} one {Delivery for following item is unavailable using the selected ZIP Code.} other {Delivery for following items are unavailable using the selected ZIP Code.}}",
  "omnishipping.unavailableDeliveryAvailablePickup": "{items, plural, =0 {} one {Delivery is unavailable for the following item. You can pick it up in a pickup point.} other {Delivery is unavailable for the following items. You can pick them up in a pickup point.}}",
  "omnishipping.optionPackagesDetails": "{items, plural, =0 {...} one {{shippingEstimate}} other {{items} packages in different terms}}",
  "omnishipping.onePackageIn": "One package in",
  "omnishipping.unavailableDeliveryOptionsPostalCode": "Change the ZIP Code or {removeFromCart}",
  "omnishipping.unavailableDeliveryOptions": "Choose another pickup point or {removeFromCart}",
  "omnishipping.changePostalCode": "Change ZIP Code",
  "omnishipping.removeFromCart": "{items, plural, =0 {} one {remove item from the cart.} other {remove items from the cart.}}",
  "omnishipping.pickUpItems": "{items, plural, =0 {} one {Pick up item} other {Pick up items}}",
  "omnishipping.allItemsPickup": "Only the Pickup in Point option is available for your items",
  "omnishipping.noTimesAvailable": "There is no appointment time available for this date.",
  "omnishipping.productsFromSeller": "Products from {seller}",
  "omnishipping.fillDeliveryData": "Fill shipping data",
  "omnishipping.deliverySplitDisclaimer": "Some items cannot be delivered to this address. Select a pickup point.",
  "omnishipping.deliverySplitDisclaimerLogged": "Some items are unavailable for your address. Take them out at one point.",
  "omnishipping.pickupSplitDisclaimer": "Some of items are unavailable for pickup in this point. Fill shipping data to deliver them.",
  "omnishipping.pickupSplitDisclaimerLogged": "Some of items are unavailable for shipping in this point. Fill shipping data to deliver them.",
  "omnishipping.price": "{value, plural, =0 {Free} other {{formattedPrice}}}",
  "omnishipping.of": "of",
  "omnishipping.askGeolocation": "Search an address with your location",
  "omnishipping.cheapestOption": "Cheapest",
  "omnishipping.fastestOption": "Quickest",
  "omnishipping.combinedOption": "Combined",
  "omnishipping.yourPackages": "Packages and delivery dates",
  "omnishipping.pickupSomeItems": "1. Pickup in point options",
  "omnishipping.receiveRemainingPackages": "2. Shipping options for remaining products",
  "omnishipping.askGeolocationTitle": "We'll suggest the pickup point closest to you",
  "omnishipping.askGeolocationSubtitle": "For that, we need your location",
  "omnishipping.askGeolocationAccept": "Use my current location",
  "omnishipping.askGeolocationManual": "Search for an address manually",
  "omnishipping.askGeolocationDenied": "Search for an address",
  "omnishipping.askGeolocationDeniedTitle": "Find nearby pickup points",
  "omnishipping.askGeolocationDeniedSubtitle": "Search for addresses that you frequent and we will locate the nearby pickup points",
  "omnishipping.askGeolocationWaiting": "Waiting on your authorization",
  "omnishipping.askGeolocationInstructions": "Select \"Allow\" in the prompt on your browser",
  "omnishipping.askGeolocationSearching": "Searching for your location",
  "omnishipping.shippingEstimate-db": "Business days",
  "omnishipping.backToAddressList": "Back to Address list",
  "omnishipping.jsError": "Something went wrong",
  "omnishipping.reloadThePage": "Reload the page",
  "omnishipping.errorId": "ErrorId: {errorId}",
  "omnishipping.noAddressTitle": "Address data not filled in",
  "omnishipping.noAddressCTA": "To add an address, {action}.",
  "omnishipping.noAddressAction": "confirm your identity"
}
