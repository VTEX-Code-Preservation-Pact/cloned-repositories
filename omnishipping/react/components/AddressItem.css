.addressItemOption {
  display: flex;
  align-items: center;
  background: #fff;
  border-bottom: 1px solid #eee;
  padding: 10px;
  box-shadow: 0 1px 1px 0 rgba(204, 204, 204, 0.5);
}

.addressItemOption:last-child {
  border-bottom: none;
}

.addressItemOption:hover,
.active {
  background: #f2f6fb;
}

.addressItemIcon {
  flex: none;
  display: flex;
  margin-right: 10px;
}

.addressItemText {
  flex: auto;
}

.addressItemTextLabel {
  font-weight: 500;
}

.sep {
  margin: 0 8px;
}

.addressItemRadio {
  display: none;
}

.text :global .state-dash,
.text :global .country-dash,
.text :global .country,
.text :global .postalCode,
.text :global .address-summary br:last-of-type,
.text :global .postal-code {
  display: none;
}
