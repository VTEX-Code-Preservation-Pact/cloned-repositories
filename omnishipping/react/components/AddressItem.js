import React, { Component } from 'react'
import PropTypes from 'prop-types'
import isEqual from 'lodash/isEqual'
import omit from 'lodash/omit'
import { connect } from 'react-redux'
import { selectAddressItem } from '../actions/order-form-actions'
import { editAddress } from '../actions/component-activity-actions'
import classNames from 'classnames/dedupe'
import RadioSelected from '../assets/components/RadioSelected'
import Radio from '../assets/components/Radio'
import { components } from 'vtex.address-form'
const { AddressSummary } = components

import styles from './AddressItem.css'

class AddressItem extends Component {
  handleClick = () => {
    const {
      address,
      editAddress,
      selectedAddress,
      selectAddressItem,
    } = this.props

    const alreadySelected = isEqual(
      omit(selectedAddress, 'addressQuery'),
      address
    )

    selectAddressItem(address, alreadySelected)
    editAddress(false)
  }

  render() {
    const { address, isSelected, rules, giftRegistryData } = this.props

    return !giftRegistryData ? (
      <label
        className={classNames('address-item', styles.addressItemOption, {
          [styles.active]: isSelected,
        })}
        onClick={this.handleClick}
        selected={isSelected}>
        <div className={`${styles.addressItemIcon} shp-option-icon`}>
          {isSelected ? <RadioSelected /> : <Radio />}
        </div>
        <div className={`shp-option-text ${styles.addressItemText}`}>
          <AddressSummary address={address} rules={rules} />
        </div>
      </label>
    ) : (
      <label
        className={classNames('address-item', styles.addressItemOption, {
          [styles.active]: isSelected,
        })}
        onClick={this.handleClick}
        selected={isSelected}>
        <div className={`${styles.addressItemIcon} shp-option-icon`}>
          {isSelected ? <RadioSelected /> : <Radio />}
        </div>
        <div className={`shp-option-text ${styles.addressItemText}`}>
          <p>{giftRegistryData.giftRegistryTypeName}</p>
        </div>
      </label>
    )
  }
}

AddressItem.propTypes = {
  address: PropTypes.object.isRequired,
  editAddress: PropTypes.func.isRequired,
  giftRegistryData: PropTypes.object,
  isSelected: PropTypes.bool,
  rules: PropTypes.object,
  selectAddressItem: PropTypes.func.isRequired,
  selectedAddress: PropTypes.object,
}

export default connect(
  null,
  {
    selectAddressItem,
    editAddress,
  }
)(AddressItem)
