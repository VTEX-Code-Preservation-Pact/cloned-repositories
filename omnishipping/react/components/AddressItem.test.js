import React from 'react'
import { Provider } from 'react-redux'
import renderer from 'react-test-renderer'
import { IntlProvider } from 'react-intl'
import { mount } from 'enzyme'
import AddressItem from './AddressItem'
import messages from '../../messages/pt.json'

describe('AddressItem', () => {
  let state, store, props

  const address = {
    abbr: 'BR',
    addressId: 'TESTE',
    addressType: 'residential',
    city: 'Rio de Janeiro',
    complement: '100',
    country: 'BRA',
    geoCoordinates: [],
    neighborhood: 'Tijuca',
    number: '',
    postalCode: '333333333',
    receiverName: 'bananinha',
    reference: null,
    state: 'RJ',
    street: 'Rua Andrade Neves',
    summary: [
      [
        {
          name: 'street',
        },
        {
          delimiter: ' ',
          name: 'number',
        },
        {
          delimiter: ', ',
          name: 'complement',
        },
      ],
      [
        {
          name: 'neighborhood',
        },
        {
          delimiter: ' - ',
          name: 'city',
        },
        {
          delimiter: ' - ',
          name: 'state',
        },
      ],
      [
        {
          name: 'postalCode',
        },
      ],
    ],
  }

  beforeEach(() => {
    state = {
      orderForm: {
        shippingData: {
          availableAddresses: [
            {
              addressId: 'TESTE',
              addressType: 'residential',
              city: 'Rio de Janeiro',
              complement: '100',
              country: 'BRA',
              geoCoordinates: [],
              neighborhood: 'Tijuca',
              number: '',
              postalCode: '333333333',
              receiverName: 'bananinha',
              reference: null,
              state: 'RJ',
              street: 'Rua Andrade Neves',
            },
            {
              addressId: 'TESTE 2',
              addressType: 'residential',
              city: 'São Paulo',
              complement: '200',
              country: 'BRA',
              geoCoordinates: [],
              neighborhood: 'Grajaú',
              number: '100',
              postalCode: '999999999',
              receiverName: 'bananinha 2',
              reference: null,
              state: 'SP',
              street: 'Rua Grajaú',
            },
          ],
          selectedAddresses: [address],
        },
      },
    }
    store = {
      subscribe: jest.fn(),
      dispatch: jest.fn(),
      getState: jest.fn(() => state),
    }
    props = {
      address: address,
      selectAddressItem: jest.fn(),
      rules: address,
    }
  })

  it('should render AddressItem and components', function() {
    const wrapper = renderer
      .create(
        <Provider store={store}>
          <IntlProvider
            locale="pt"
            messages={{ ...messages, ...{ 'country.BRA': 'BRA' } }}>
            <AddressItem {...props} />
          </IntlProvider>
        </Provider>
      )
      .toJSON()
    expect(wrapper).toMatchSnapshot()
  })

  it('should trigger selectAddressItem if click address-item', done => {
    const mockChangeSelectedAdressItem = jest.fn()
    const props = {
      address: address,
      selectAddressItem: mockChangeSelectedAdressItem,
      rules: address,
    }
    const wrapper = mount(
      <Provider store={store}>
        <IntlProvider
          locale="pt"
          messages={{ ...messages, ...{ 'country.BRA': 'BRA' } }}>
          <AddressItem {...props} />
        </IntlProvider>
      </Provider>
    )

    const locateItem = wrapper.find('.address-item')

    locateItem.simulate('click')
    process.nextTick(() => {
      try {
        wrapper.update()
        expect(mockChangeSelectedAdressItem).toHaveBeenCalledWith(
          props.selectedAddress[0]
        )
        expect(mockChangeSelectedAdressItem.mock.calls).toHaveLength(1)
      } catch (e) {
        return done(e)
      }
      done()
    })
  })
})
