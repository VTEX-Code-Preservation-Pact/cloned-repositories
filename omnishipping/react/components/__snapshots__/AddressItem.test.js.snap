// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`AddressItem should render AddressItem and components 1`] = `
<label
  className="address-item addressItemOption"
  onClick={[Function]}
  selected={undefined}
>
  <div
    className="addressItemIcon shp-option-icon"
  >
    <svg
      className="svg"
      height="16"
      viewBox="0 0 16 16"
      width="16"
      xmlns="http://www.w3.org/2000/svg"
    >
      <path
        d="M8 0C3.584 0 0 3.584 0 8s3.584 8 8 8 8-3.584 8-8-3.584-8-8-8zm0 14.4A6.398 6.398 0 0 1 1.6 8c0-3.536 2.864-6.4 6.4-6.4 3.536 0 6.4 2.864 6.4 6.4 0 3.536-2.864 6.4-6.4 6.4z"
        fill="#666"
      />
    </svg>
  </div>
  <div
    className="shp-option-text addressItemText"
  >
    <div
      className="address-summary address-summary-BRA"
    >
      <span>
        <span
          className="street"
        >
          Rua Andrade Neves
        </span>
      </span>
      <span>
        
        <span
          className="complement"
        >
          100
        </span>
      </span>
      <br
        className="line1-delimiter"
      />
      <span>
        <span
          className="neighborhood"
        >
          Tijuca
        </span>
      </span>
      <span>
        <span
          className="city-delimiter"
        >
           - 
        </span>
        <span
          className="city"
        >
          Rio de Janeiro
        </span>
      </span>
      <span>
        <span
          className="state-delimiter"
        >
           - 
        </span>
        <span
          className="state"
        >
          RJ
        </span>
      </span>
      <br
        className="line2-delimiter"
      />
      <span>
        <span
          className="postalCode"
        >
          333333333
        </span>
      </span>
      <br
        className="line3-delimiter"
      />
      <span
        className="country"
      >
        BRA
      </span>
    </div>
  </div>
</label>
`;
